src/ape/api/providers.py

```diff
@@ -926,7 +926,10 @@ def _prepare_call(self, txn: TransactionAPI, **kwargs) -> List:
             if value is not None and not isinstance(value, str):
                 txn_dict[field] = to_hex(value)
 
-        block_identifier = kwargs.pop("block_identifier", "latest")
+        if "block_identifier" in kwargs:
+            block_identifier = hex(kwargs["block_identifier"])
```

Comment:
What if the user already passed a hex str?
Also, if the user explicitly passes `"latest"`, this will fail and try to convert `"latest"` to a hex str.

Maybe it should be like this:

```python
from eth_utils import to_hex, is_0x_prefixed
...

block_identifier = kwargs.get("block_identifier", "latest")
if isinstance(block_identifier, int):
    block_identifier = to_hex(block_identifier])

# else: assume the user specified a correct HexStr or special-value
```